src/App.tsx

```diff
@@ -1,61 +1,42 @@
+import React, { useState } from 'react';
 import './App.scss';
+import { TodoList } from './components/TodoList';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { getVisibleToDos } from './servises/VisibleTodos';
+import { Form } from './components/Form';
+import { getNewTodoId } from './servises/NewTodoId';
+import { FieldsForm } from './types/FieldsForm';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+
+const visibleToDos = getVisibleToDos(usersFromServer, todosFromServer);
+
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState(visibleToDos);
+
+  const addNewTodo = ({ title, selectUser }: FieldsForm) => {
+    const newTodo = {
+      id: getNewTodoId(todos),
+      title,
+      completed: false,
+      userId: selectUser,
+      user: usersFromServer
+        .find((user: User) => user.id === selectUser) || null,
+    };
+
+    setTodos((currentTodos: Todo[]) => [...currentTodos, newTodo]);
+  };
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
-
-export const App = () => {
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
-        <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
-        </div>
-
-        <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>Choose a user</option>
-          </select>
-
-          <span className="error">Please choose a user</span>
-        </div>
-
-        <button type="submit" data-cy="submitButton">
-          Add
-        </button>
-      </form>
-
-      <section className="TodoList">
-        <article data-id="1" className="TodoInfo TodoInfo--completed">
-          <h2 className="TodoInfo__title">
-            delectus aut autem
-          </h2>
-
-          <a className="UserInfo" href="mailto:Sincere@april.biz">
-            Leanne Graham
-          </a>
-        </article>
-
-        <article data-id="15" className="TodoInfo TodoInfo--completed">
-          <h2 className="TodoInfo__title">delectus aut autem</h2>
-
-          <a className="UserInfo" href="mailto:Sincere@april.biz">
-            Leanne Graham
-          </a>
-        </article>
-
-        <article data-id="2" className="TodoInfo">
-          <h2 className="TodoInfo__title">
-            quis ut nam facilis et officia qui
-          </h2>
+      <Form
+        onAdd={addNewTodo}
+      />
```

Comment:
```suggestion
      <Form onAdd={addNewTodo} />
```

Here we have only one prop, so can leave it as it is